src/vcpkg/commands.applocal.cpp

```diff
@@ -0,0 +1,537 @@
+                {
+                    deploy_binary(target_binary_dir, bin_dir, lib);
+                }
+
+                deployPluginsQt("scenegraph", target_binary_dir, qt_plugins_dir);
+                deployPluginsQt("qmltooling", target_binary_dir, qt_plugins_dir);
+            }
+
+            std::error_code ec;
+            std::vector<Path> children = m_fs.get_files_non_recursive(target_binary_name, ec);
+            for (auto&& c : children)
+            {
+                if (c == "Qt5Declarative.dll" || c == "Qt5Declaratived.dll")
+                {
+                    deployPluginsQt("qml1tooling", target_binary_dir, qt_plugins_dir);
+                }
+
+                if (c == "Qt5Positioning.dll" || c == "Qt5Positioningd.dll")
+                {
+                    deployPluginsQt("position", target_binary_dir, qt_plugins_dir);
+                }
+
+                if (c == "Qt5Location.dll" || c == "Qt5Locationd.dll")
+                {
+                    deployPluginsQt("geoservices", target_binary_dir, qt_plugins_dir);
+                }
+
+                if (c == "Qt5Sensors.dll" || c == "Qt5Sensorsd.dll")
+                {
+                    deployPluginsQt("sensors", target_binary_dir, qt_plugins_dir);
+                    deployPluginsQt("sensorgestures", target_binary_dir, qt_plugins_dir);
+                }
+
+                if (c == "Qt5WebEngineCore.dll" || c == "Qt5WebEngineCored.dll")
+                {
+                    deployPluginsQt("qtwebengine", target_binary_dir, qt_plugins_dir);
+                }
+
+                if (c == "Qt53DRenderer.dll" || c == "Qt53DRendererd.dll")
+                {
+                    deployPluginsQt("sceneparsers", target_binary_dir, qt_plugins_dir);
+                }
+
+                if (c == "Qt5TextToSpeech.dll" || c == "Qt5TextToSpeechd.dll")
+                {
+                    deployPluginsQt("texttospeech", target_binary_dir, qt_plugins_dir);
+                }
+
+                if (c == "Qt5SerialBus.dll" || c == "Qt5SerialBusd.dll")
+                {
+                    deployPluginsQt("canbus", target_binary_dir, qt_plugins_dir);
+                }
+            }
+        }
+
+        bool deploy_binary(const Path& target_binary_dir,
+                           const Path& installed_dir,
+                           const std::string& target_binary_name)
+        {
+            const auto source = installed_dir / target_binary_name;
+            const auto target = target_binary_dir / target_binary_name;
+
+            const auto mutant_name = Strings::to_utf16("vcpkg-applocal-" + Hash::get_string_sha256(target_binary_dir));
+
+            HANDLE the_mutant = ::CreateMutexW(nullptr, FALSE, mutant_name.c_str());
+
+            if (the_mutant == NULL)
+            {
+                Checks::exit_with_message(VCPKG_LINE_INFO, "Failed\n");
+            }
+
+            WaitForSingleObject(the_mutant, INFINITE);
```

Comment:
This function is a bit complicated to manage the Mutant lifetime without RAII. With this suggestion, you can remove the Release/Close lines before the return statements.

```suggestion
            const auto mutant_name = "vcpkg-applocal-" + Hash::get_string_sha256(target_binary_dir);

            struct MutantGuard {
                MutantGuard(StringView name) {
                    h = ::CreateMutexW(nullptr, FALSE, Strings::to_utf16(name).c_str());
                    if (h) {
                        WaitForSingleObject(h, INFINITE);
                    }
                }
                explicit operator bool() const { return h; }
                ~MutantGuard() {
                    if (h) {
                        ReleaseMutex(h);
                        CloseHandle(h);
                    }
                }
                HANDLE h;
            };
            const MutantGuard mutant(mutant_name);

            if (!mutant)
            {
                Checks::exit_with_message("Failed to acquire mutant %s", mutant_name);
            }
```